kratos/utilities/data_type_traits.h

```diff
@@ -310,7 +310,7 @@ template<class TDataType> class DataTypeTraits
         if constexpr(TCheckIndex == 0) {
             return false;
         } else {
-            static_assert(TCheckIndex != TCheckIndex, "Invalid dimension index.");
+            static_assert(false, "Invalid dimension index.");
```

Comment:
this will always fail

the reason we added smth as ridiculus as `TCheckIndex != TCheckIndex` is because the predicate in `static_assert` must depend on the provided template parameters, otherwise compilers (or at least GCC) will evaluate it regardless of whether the `else` branch is ever taken.

Reply:
Btw the point was to issue a compile-time error message if the `constexpr` `else` branch is taken. Let @sunethwarna and me know if you have a better solution for that than `static_assert` because we use this pattern elsewhere too.

Reply:
Ok I will find another way to check that, but seems that now comparing agains the same variable is a Warning, so error in the ci